src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -0,0 +1,212 @@
+        {
+            get => _fingerprintPhrase;
+            set
+            {
+                SetProperty(ref _fingerprintPhrase, value);
+            }
+        }
+
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                SetProperty(ref _email, value);
+            }
+        }
+
+        protected override II18nService i18nService => _i18nService;
+        protected override IEnvironmentService environmentService => _environmentService;
+        protected override IDeviceActionService deviceActionService => _deviceActionService;
+        protected override IPlatformUtilsService platformUtilsService => _platformUtilsService;
+
+        public void StopCheckLoginRequestStatus()
+        {
+            try
+            {
+                _checkLoginRequestStatusCts?.Cancel();
+                _checkLoginRequestStatusCts?.Dispose();
+            }
+            catch (Exception ex)
+            {
+                _logger.Exception(ex);
+            }
+        }
+
+        public void StartCheckLoginRequestStatus()
+        {
+            try
+            {
+                _checkLoginRequestStatusCts?.Cancel();
+                _checkLoginRequestStatusCts = new CancellationTokenSource();
+                _checkLoginRequestStatusTask = new TimerTask(_logger, CheckLoginRequestStatus, _checkLoginRequestStatusCts).RunPeriodic(TimeSpan.FromSeconds(REQUEST_TIME_UPDATE_PERIOD_IN_SECONDS));
+            }
+            catch (Exception ex)
+            {
+                _logger.Exception(ex);
+            }
+        }
+
+        private async Task CheckLoginRequestStatus()
+        {
+            if (string.IsNullOrEmpty(_requestId) || string.IsNullOrEmpty(_requestAccessCode))
+            {
+                return;
+            }
+
+            try
+            {
+                var response = await _authService.GetPasswordlessLoginResponse(_requestId, _requestAccessCode);
+
+                if (!response.RequestApproved)
+                {
+                    return;
+                }
+
+                StopCheckLoginRequestStatus();
+
+                var decKey = await _cryptoFunctionService.RsaDecryptAsync(response.Key, _requestKeyPair.Item2);
+                var decPasswordHash = await _cryptoFunctionService.RsaDecryptAsync(response.MasterPasswordHash, _requestKeyPair.Item2);
+
+                var authResult = await _authService.LogInPasswordlessAsync(Email, _requestAccessCode, _requestId, decKey, decPasswordHash);
+                await AppHelpers.ResetInvalidUnlockAttemptsAsync();
+
+                if (authResult.CaptchaNeeded)
+                {
+                    if (await HandleCaptchaAsync(authResult.CaptchaSiteKey))
+                    {
+                        await CheckLoginRequestStatus();
+                        _captchaToken = null;
+                    }
+                    return;
+                }
+                _captchaToken = null;
```

Comment:
I don't like much setting base fields here when it can be donde above. Move this to a method on the base class, so here you can have:
```c#
if(!await HandleCaptchaAsync(authResult.CaptchaNeeded, authResult.CaptchaSiteKey, CheckLoginRequestStatus))
{
     return;
}
```